### PR TITLE
Basic BS tab

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# Basic BS tab

Required for https://app.shortcut.com/oazo-apps/story/3869/basic-buy-form and other basic BS related stories.
  
## Changes 👷‍♀️

Created a new tab for Optimization for vaults. Currently the tab does not have any actual content. Component structure was based on Protection tab and contains some empty components, but they are going to be needed in the future.

image
  
## How to test 🧪

Go to any vault and see if Optimization tab is available.

### CODE REVIEW

File: components/VaultTabSwitch.tsx [hunk 1]
```diff
@@ -128,6 +131,7 @@ export function VaultTabSwitch({
   const { uiChanges } = useAppContext()
   const { t } = useTranslation()
   const newComponentsEnabled = useFeatureToggle('NewComponents')
+  const basicBSEnabled = useFeatureToggle('BasicBS')
```
Comment (piekczyk) [inline components/VaultTabSwitch.tsx:134]: we will most likely release BS and BB in the same time so this feature toggle name may be misleading

File: features/automation/optimization/controls/OptimizationDetailsControl.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+function renderLayout({ vault }: { vault: Vault }) {
+  return <OptimizationDetailsLayout vault={vault} />
+}
+
+interface OptimizationDetailsControlProps {
+  vault: Vault
+}
+
+export function OptimizationDetailsControl({ vault }: OptimizationDetailsControlProps) {
+  return renderLayout({ vault })
```
Comment (piekczyk) [inline features/automation/optimization/controls/OptimizationDetailsControl.tsx:15]: I know that we have this pattern in stop loss (Adam's idea) but IMO render layout method add unnecessary complexity here.

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: components/VaultTabSwitch.tsx (hunk 1 above)
Comment (piotrkonowrocki) [reply components/VaultTabSwitch.tsx:134]: `BasicBs` -> `Basic Buy & Sell`, at least in my mind that's what I meant, `B` in `BS` stands for `Buy`, not `Basic` 😄 What would you suggest to call it?

File: features/automation/optimization/controls/OptimizationDetailsControl.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply features/automation/optimization/controls/OptimizationDetailsControl.tsx]: Removed.

**Review by piekczyk — APPROVED** [review]

**Pull request merged**